### DISCUSSION TITLE
Collegarsi da remoto con rete mobile

### DISCUSSION THREAD

**Discussion by Fortemartello** [indomus/forum#980] [HUB: Home Assistant]
Salve, ho iniziaiato da poco avventura in campo domotico e la stò trovando affascinante. Normalmente cerco di risolvere da me quasiasi problema ma per questo non sono riuscito ancora a trovare una soluzione adatta a me.
PROBLEMA: quello che vorrei è collegarmi ad home assistant dall'esterno tramite rete mobile iliad (ora ho quella, mi vanno bene anche altri operatori, l'unico limite è il prezzo 10/15 euro al mese)
CONFIGURAZIONE: sim su router 4g lte (devo ancora scegliere quale comprare), raspberry pi4 con raspbian e home assistant core installato su docker

Attendo vostre risposte, grazie.
Lorenzo

**Comment by sky7176** [comment]
Ciao Lorenzo,
non ho capito una cosa, ti vuoi collegare da remoto al tuo HA, oppure il tuo HA deve usare il router LTE?

Giuseppe

**Reply by Fortemartello** [reply]
Ciao Giuseppe, vorrei accedere ad HA dall'esterno tramite la sua app HA companion. Raspberry si collega al momento ad un ruter provvisorio (non LTE) attraverso cavo ethernet. La connessione ad internet avviene tramite tethering dal mio cellulare.

**Comment by sky7176** [comment]
Ciao,
devi configurare un inoltro porte della porta 8123 verso l'indirizzo della tua raspberry, questo inoltro, o port forwording, lo configuri sul router.
Dopo devi avere anche un servizio di DynamicDNS attivo in modo da associare il tuo ip pubblico, che sicuramente sarà dinamico, ad un nome di dominio facilmente ricordabile.
In sostanza:
1) sul tuo router dici che tutto il traffico internet in arrivo sulla porta 8123 deve essere "ruotato" verso l'indirizzo IP della tua raspberry, es 192.168.1.2
2) attivare sulla raspberry l'addon Duck DNS in modo da associare un nome mmemonico es. casamia.duckdns.org

Fatte queste due cose installi l'app sul cellulare e la configuri per accedere a: casamia.duckdns.org:8123, se hai utilizzato Duck DNS oppure a qualche cosa di simile se hai utilizzato altri servizi DDNS.

La medesima cosa la potresti fare anche dal router, abitualmente hanno la sezione DynamicDNS.

Per esempio io entrambi i punti li faccio con il firewall che ho a casa e per il servizio DDNS utilizzo noip.com attivo sul firewall, di tale servizio ne esiste una versione a pagamento ed una versione free.

La differenza tra la versione a pagamento e quella free è nel fatto che quest'ultima permette un solo dominio e almeno una volta al mese ti devi collegare al loro sito, altrimenti il servizio scade.

Per rendere un po più sicura la cosa, dal punto di vista della sicurezza informatica, potresti configurare l'app sul cellulare non per comunicare sulla 8123, ma per esempio sulla 9000, poi al router dici di ruotare tutto il traffico in ingresso dalla porta 9000 verso la porta 8123 della raspberry, in questo modo, se un maleintenzionato dovesse fare uno scan port troverebbe la porta 9000 aperta e non essendo la porta 9000 ufficialmente assegnata ad uno specifico servizio, per es. la 80 che è quella di http oppute la 443 per https, avrebbe, il condizionale è d'obbligo, difficoltà a capire cosa ci sia in ascolto dall'altra parte e questo gli renderebbe la vita un attimo più complicata.

Ma questa è una cosa da sistemisti paranoici della sicurezza.

Direi che per il momento potresti fermarti ai due punti che ti ho indicato.

In ogni caso se hai bisogno sono qui.

Giuseppe

**Reply by Fortemartello** [reply]
Grazie della risposta Giuseppe.
Questa soluzione mi andrebbe bene se avvessi linea fissa con ip pubblico. Tuttavia avendo sim iliad per la connessione l'ip è nattato (quindi non pubblico). Questo impedisce qualsiasi port forwording.
Questo è quello che ho capito cercando online. Dimmi pure se sbaglio.

Lorenzo

**Reply by snakuzzo** [reply]
Guardati ngrok e/o remote.it

**Reply by Niconaco** [reply]
Ciao, ho una connessione nuova 4g con verymobile, quindi è una rete nat, volevo chiedere se qualcuno mi potrebbe aiutare gentilmente per configurare il mio server debian o comunque home assistant con remote.it, non riesco a trovare nessuna guida che spiega come fare... Grazie mille.

**Reply by snakuzzo** [reply]
Prova a seguire la guida recente che abbiamo scritto per creare un server VPN (gratuito) su una VM su GCP

**Reply by Niconaco** [reply]
riusciresti a mandarmi il link che non lo trovo? Grazie mille

**Reply by snakuzzo** [reply]
Crezione VM -> [QUI](https://indomus.it/guide/istanziare-una-vm-su-google-cloud-platform-gcp-per-uso-personale/)
Collegarsi da remoto -> [QUI](https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-via-reti-provider-che-usino-nat/)

**Reply by Niconaco** [reply]
Ci avevo già provato ma creando la vm mi dava come costo stimato 6 dollari facendo tutto uguale... Non so se è normale...

**Reply by snakuzzo** [reply]
Leggi bene la guida...e leggi bene il free-tier di Google (l'abbiamo scritto).
Quello che vedi alla fine non è il costo effettivo "se rispetti le regole del free-tier" [QUI](https://cloud.google.com/free/docs/gcp-free-tier?hl=it&_ga=2.192441513.-2064091823.1638186224&_gac=1.254252538.1638186224.Cj0KCQiAkZKNBhDiARIsAPsk0WijnnbNBxhQVA4BvzYhQamGWT2484ReqmMQDGIHSuiUaavYL8Hn3YsaAlGKEALw_wcB).
L'abbiamo scritto pure con una nota in rosso :D 
Io ho la VM attiva da mesi e non ho mai pagato un centesimo. Tutti i mesi arriva la fattura di Google da $0.00

**Reply by Niconaco** [reply]
va bene grazie mille!

**Comment by sky7176** [comment]
Lorenzo,
mi spiace ma non conosco la configurazione di iliad, ma se nattato la vedo dura se non mettere di mezzo un server vpn.

Giuseppe

**Comment by KC-inDomus** [comment]
> Salve, ho iniziaiato da poco avventura in campo domotico e la stò trovando affascinante. Normalmente cerco di risolvere da me quasiasi problema ma per questo non sono riuscito ancora a trovare una soluzione adatta a me.
> 
> PROBLEMA: quello che vorrei è collegarmi ad home assistant dall'esterno tramite rete mobile iliad (ora ho quella, mi vanno bene anche altri operatori, l'unico limite è il prezzo 10/15 euro al mese)
> 
> CONFIGURAZIONE: sim su router 4g lte (devo ancora scegliere quale comprare), raspberry pi4 con raspbian e home assistant core installato su docker
> 
> 
> 
> Attendo vostre risposte, grazie.
> 
> Lorenzo
> 
> 

[questa](https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-via-reti-provider-che-usino-nat/) l'hai letta?
